### ISSUE TITLE
Smart alarms on Mi Band 2

### ISSUE THREAD

**Issue by 6arms1leg** [Freeyourgadget/Gadgetbridge#471]
As we discussed [here](323#issuecomment-256625747) (#323), the smart alarm feature on the Mi Band 2 once worked, but seems to be officially dropped by Xiaomi on newer firmware.
@cpfeiffer, just in case you want to implement it anyway, here is a BT snoop log of setting a smart alarm to 0AM on a Mi Band 2 with the Mi Band Tools app:

[btsnoop_hci-alarm_smart_00am.zip](https://github.com/Freeyourgadget/Gadgetbridge/files/671232/btsnoop_hci-alarm_smart_00am.zip)

However, maybe it would be better to just disable the smart alarm feature completely for the Mi Band 2 in GB, so nobody gets confused (or at least add a small description that it might not work).

**Comment by cpfeiffer** [comment]
Thanks. I'm not sure if your log really contains the necessary commands, though. It sets the current time, then configures the (first) alarm at 0:00 AM with repetition on every. Then it configures some BLE connection things.

You configure an alarm in Gadgetbridge the very same way.

Does this actually work for you? Either it does not work, or there is some other communication that is not captured in your log.

I agree that we should not display the "Smart alarm" checkbox when not supported.

**Comment by 6arms1leg** [comment]
Oh, sorry if I wasted your time!
On the device I used, smart alarms didn't work but I thought the BLE commands for setting smart alarms would be sent to the Mi Band 2 anyway.
Maybe the Mi Band Tools app checks the firmware of the device (and hence wheather smart alarms are supported) before sending the alarm configuration to the device (in which case the smart alarm option shouldn't be available anyway) or maybe the smart alarm option is a fake.
